### DISCUSSION TITLE
Chromatic matching

### DISCUSSION THREAD

**Discussion by chanceant** [xsuite/xsuite#455] [Ideas]
Dear all,

Thank you for the great job you are doing.
I am new with XSuite and I am examining the possibility to directly use XSuite instead of Mad-X.
That is already great to get the chromatic functions ax, bx, and wx with the twiss method.
Unfortunately,  that is not possible to use them as targets for matching modules (only first order functions are currently allowed).
Is it foreseen to add this functionality for TargetSet?

Other suggestion: the second-order dispersion is not currently calculated, even with twiss.
That would be very useful to have this functionality, especially for the interaction regions.

Thank you in advance for your answer,

**Comment by giadarol** [comment]
Hello!
The chromatic function can already be used as match targets (with a slightly lengthier syntax). 
You need to do something like:
```python
xt.Target(at="ip3", tar="bx_chrom", value=0),
```

We are already planning to add second order chromaticity and second order dispersion, coming soon...

**Comment by giadarol** [comment]
@chanceant Hello, a new version of the xsuite packages was released yesterday. It provides second order dispersion and second order chromaticity in the results of the twiss. It also allows using chromatic properties as target keywords in match.

Don't hesitate to reopen, in case you still miss something or you you face other issues for your optics design studies.
